loadtest/user/userentity/actions.go

```diff
@@ -697,6 +698,60 @@ func (ue *UserEntity) GetChannelMembers(channelId string, page, perPage int) err
 	return ue.store.SetChannelMembers(channelMembers)
 }
 
+// GetAllChannelMembersForUser gets all channel memberships for the
+// specified user regardless of the team the channels are part of
+func (ue *UserEntity) GetAllChannelMembersForUser(userId string) error {
+	serverVersion := ue.Store().ServerVersion()
+	// Versions 10.9.0 and later support streaming all the members by setting
+	// page to -1, so there is no need to paginate
+	if serverVersion.GTE(semver.MustParse("10.9.0")) {
+		membersWithTeamData, _, err := ue.client.GetChannelMembersWithTeamData(context.Background(), userId, -1, 0)
```

Comment:
I suppose that from a client-side perspective, we aren't truly streaming anything since this will return the whole list?

Perhaps there could be a way to leverage the new iterators concept in Go 1.23+ and actually stream them through.

Reply:
The client needs the entire data all at once. So there isn't really an opportunity to stream inside the client.